Maple2.Server.Game/Model/Field/Buff.cs

```diff
@@ -93,42 +107,26 @@ public class Buff : IUpdatable, IByteSerializable {
         return enabled;
     }
 
-    public void Remove() {
-        if (Owner.Buffs.Remove(Id, out _)) {
-            field.Broadcast(BuffPacket.Remove(this));
-        }
+    public void Disable() {
+        Enabled = false;
+        canProc = false;
     }
+    public void Enable() => Enabled = true;
 
-    public virtual void Update(long tickCount) {
-        if (!activated) {
-            Activate();
-        }
-
-        if (canExpire && !canProc && tickCount > EndTick) {
-            Remove();
-            return;
-        }
-
-        // TODO: Check conditions less frequently
-        if (!UpdateEnabled()) {
-            return;
-        }
-
+    public void Proc(long tickCount) {
```

Comment:
why rename this from `Update`, seems clearer to keep these things with `IUpdatable`. A proc isn't guaranteed on every call of this function anyways (`tickCount < NextProcTick`)